tools/serverpod_cli/lib/src/util/protocol_helper.dart

```diff
@@ -0,0 +1,46 @@
+import 'dart:io';
+
+import 'package:serverpod_cli/src/config/config.dart';
+import 'package:path/path.dart';
+
+class ProtocolSource {
+  String yaml;
+  Uri uri;
+  List<String> protocolRootPathParts;
+
+  ProtocolSource(this.yaml, this.uri, this.protocolRootPathParts);
+}
+
+class ProtocolHelper {
+  static Future<List<ProtocolSource>> loadProjectYamlProtocolsFromDisk(
+    GeneratorConfig config,
+  ) async {
+    var sourceDir = Directory(joinAll(config.protocolSourcePathParts));
+    var sourceFileList = await sourceDir.list(recursive: true).toList();
+    sourceFileList.sort((a, b) => a.path.compareTo(b.path));
```

Comment:
Question: Why is this sort required?

Reply:
I don't think it is, legacy that I haven't removed :)